collector/service.go

```diff
@@ -127,68 +152,158 @@ var (
 )
 
 func (c *serviceCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
-	var dst []Win32_Service
-	q := queryAllWhere(&dst, c.queryWhereClause)
-	if err := wmi.Query(q, &dst); err != nil {
-		return nil, err
-	}
-	for _, service := range dst {
-		pid := strconv.FormatUint(uint64(service.ProcessId), 10)
+	if *wmiDisabled {
```

Comment:
just for readability i would extract both collections in different func , also it might be possible to share some code between both